### PR TITLE
Change camera::get_ray() function to int params

### PR DESCRIPTION
**Pull request by hollasch**
Text changes to follow after this is finished.

### CODE REVIEW

File: src/InOneWeekend/camera.h [hunk 1]
```diff
@@ -60,6 +56,7 @@ class camera {
     }
 
   private:
+    int image_height;
```
Comment (trevordblack) [inline src/InOneWeekend/camera.h:59]: it's kind of weird having image_width be public and image_height be private?

Consider having image_height be public and just solve for any of the image_height, image_width, aspect ratio that are not explicitly defined? ...

No. That doesn't really work. If they're all public it's hard to know which were explicitly set unless we create constructors. But that's not a fantastic idea if we let main set aspect_ratio.

What was the decision behind having aspect ratio and width be set by main? I don't remember.

We shouldn't do getters/setters (we decided no a while ago).

WDYT?

File: src/InOneWeekend/camera.h [hunk 2]
```diff
@@ -85,10 +84,9 @@ class camera {
         lens_radius = aperture / 2;
     }
 
-    ray get_ray(double s, double t) const {
-        // Return the ray from the projection point to the indicated pixel. Coordinates s,t are
-        // the normalized image-based coordinates of the pixel. Image left is s=0, image right
-        // is s=1, image top is t=0, image bottom is t=1.
```
Comment (trevordblack) [inline src/InOneWeekend/camera.h:91]: What was the decision behind removing the comments?

**Review by trevordblack — APPROVED** [review]
The fundamental change here (_i.e._ get_ray() to ints) is _fine_.

I'm not entirely sure the motivating factor behind the change. If I had to guess it's easier to express the move from 1spp to multiple with this change? If that's the case then yes, I think this is probably valuable. Not entirely certain though. If you think this is worth it then feel free to ship.

With that said, the private image_height thing has me scratching my head. Maybe we should just set width and height in main and calculate aspect? What was the reason for the way it's set up now? Goddamn we've been on this too long.

File: src/InOneWeekend/camera.h (hunk 1 above)
Comment (hollasch) [reply src/InOneWeekend/camera.h:59]: Three variables, two unknowns. The main idea is that while developing, people will typically alter parameters to adjust speed of rendering, image dimensions being one aspect. Once the render aspect ratio is done, it's typically fixed. So they can just adjust a single intuitive variable (width) or two variables plus at least a mental calculation. The former is a nicer interface. Also, if you want to compare renders without regard to size, then you want a fixed aspect ratio.

File: src/InOneWeekend/camera.h (hunk 2 above)
Comment (hollasch) [reply src/InOneWeekend/camera.h:91]: This is one of several increments to follow, so if there will be a comment, it will come after the dust settles.

**Pull request merged**